From a29f4e7cc20d679a3dc729aa5b48d8926653e792 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hmehrtens@maxlinear.com>
Date: Fri, 12 Feb 2021 17:47:59 +0100
Subject: [PATCH] PONRTSYS-7990: Fix compile warning

Fix this compile warning when CONFIG_REGMAP_ICC is not set:
drivers/char/ltq_icc/drv_icc.c:116:12: warning: 'icc_sync_write' declared 'static' but never defined [-Wunused-function]
 static int icc_sync_write(icc_msg_t *buf, icc_msg_t *retbuf);
            ^~~~~~~~~~~~~~
---
 drivers/char/ltq_icc/drv_icc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/char/ltq_icc/drv_icc.c b/drivers/char/ltq_icc/drv_icc.c
index f0a93a3e9f9c..432752096904 100644
--- a/drivers/char/ltq_icc/drv_icc.c
+++ b/drivers/char/ltq_icc/drv_icc.c
@@ -113,7 +113,6 @@ icc_fifo_t ICC_BUFFER[MAX_CLIENT];
 static struct semaphore icc_callback_sem;
 #endif
 static DEFINE_SPINLOCK(icc_sync_lock);
-static int icc_sync_write(icc_msg_t *buf, icc_msg_t *retbuf);
 
 
 #ifdef CONFIG_SOC_TYPE_GRX500_TEP
@@ -540,6 +539,8 @@ void process_icc_message(unsigned int uiClientId)
 
 
 #ifdef CONFIG_REGMAP_ICC
+static int icc_sync_write(icc_msg_t *buf, icc_msg_t *retbuf);
+
 /* regmap write function */
 int icc_regmap_sync_write(icc_devices ClientId,
 				phys_addr_t paddr, unsigned int val)
